ts/components/game/Player.tsx

```diff
@@ -44,6 +44,9 @@ class Player extends React.Component<PlayerProps, {}> {
 
 		/* Equipment */
 		var heroEntity = playEntities.filter(filterByCardType(CardType.HERO)).first();
+		if (!heroEntity) {
+			heroEntity = (this.props.entities.get(Zone.GRAVEYARD) || Immutable.Map<number, Entity>()).first();
```

Comment:
You could view it as the hero "transforming back" after being defeated. I don't think it makes much of a difference either way.